src/pages/files.tsx

```diff
@@ -150,6 +150,75 @@ const FileUploader = React.forwardRef<IFileUploaderRef, IFileUploaderProps>((pro
 
 FileUploader.displayName = 'FileUploader';
 
+/**
+ * Component for the ellipsis menu dropdown for each file.
+ */
+interface IFileMenuProps {
+  model: Contents.IModel;
+  onDownload: (model: Contents.IModel) => void;
+  onDelete: (model: Contents.IModel) => void;
+}
+
+function FileMenu(props: IFileMenuProps) {
+  const [isOpen, setIsOpen] = useState(false);
+  const menuRef = useRef<HTMLDivElement>(null);
+
+  // We'll close the menu when clicking outside the component.
+  useEffect(() => {
+    if (!isOpen) {
+      return;
+    }
+
+    const handleClickOutside = (e: MouseEvent) => {
+      if (menuRef.current && !menuRef.current.contains(e.target as Node)) {
+        setIsOpen(false);
+      }
+    };
+
+    document.addEventListener('mousedown', handleClickOutside);
+    return () => document.removeEventListener('mousedown', handleClickOutside);
+  }, [isOpen]);
+
+  const handleMenuClick = (e: React.MouseEvent) => {
+    e.stopPropagation();
+    setIsOpen(!isOpen);
+  };
+
+  const handleDownload = (e: React.MouseEvent) => {
+    e.stopPropagation();
+    props.onDownload(props.model);
+    setIsOpen(false);
+  };
+
+  const handleDelete = (e: React.MouseEvent) => {
+    e.stopPropagation();
+    props.onDelete(props.model);
+    setIsOpen(false);
+  };
+
+  return (
+    <div className="je-FileMenu" ref={menuRef}>
+      <button
+        className="je-FileMenu-trigger"
+        aria-label={`Options for ${props.model.name}`}
+        onClick={handleMenuClick}
+      >
+        {isOpen ? <EverywhereIcons.dropdownTriangle.react /> : <EverywhereIcons.ellipsis.react />}
+      </button>
+      {isOpen && (
+        <div className="je-FileMenu-dropdown">
```

Comment:
There are a few edge cases when the menu will be clipped:

<img width="429" height="362" alt="Image" src="https://github.com/user-attachments/assets/8d5877db-d98f-47a1-b294-5c77361f7f57" />

<img width="439" height="516" alt="Image" src="https://github.com/user-attachments/assets/7c42d3b7-8df6-41c6-a416-9d98dca6db2d" />

While being clipped by the warning message is mostly harmless (user can scroll down), I think we need to adjust the case when it gets clipped for tiles on the left edge.

Reply:
> I think we need to adjust the case when it gets clipped for tiles on the left edge.

Addressed in 49ee90491a022e636f48b32fca82568e38954e34